js/app/modules/contentGroup/contentGroup.js

```diff
@@ -267,6 +271,10 @@ const ContentGroup = new Module.Class({
 
     },
 
+    _on_reading_history_changed: function () {
+        this._arrangement.mark_read(ReadingHistoryModel.get_default().get_read_articles());
```

Comment:
Indeed, it was my point that this is the first property of a model that cannot be considered immutable. We've worked around it so far with the ReadingHistoryModel, but now that it needs to influence the UI instead of just the queries, it seems like it should be made more model-like.

Since a real per-user EKN metadata store is probably not realistic, how about having each card check the default ReadingHistoryModel? As far as I can see, that would make it quite easy to refactor in the right direction later.